module/zcommon/zfs_prop.c

```diff
@@ -116,6 +116,34 @@ zfs_prop_init(void)
 		{ "gzip-9",	ZIO_COMPRESS_GZIP_9 },
 		{ "zle",	ZIO_COMPRESS_ZLE },
 		{ "lz4",	ZIO_COMPRESS_LZ4 },
+		{ "lz4fast-1",	ZIO_COMPRESS_LZ4FAST_1},
+		{ "lz4fast-2",	ZIO_COMPRESS_LZ4FAST_2 },
+		{ "lz4fast-3",	ZIO_COMPRESS_LZ4FAST_3 },
+		{ "lz4fast-4",	ZIO_COMPRESS_LZ4FAST_4 },
+		{ "lz4fast-5",	ZIO_COMPRESS_LZ4FAST_5 },
+		{ "lz4fast-6",	ZIO_COMPRESS_LZ4FAST_6 },
+		{ "lz4fast-7",	ZIO_COMPRESS_LZ4FAST_7 },
+		{ "lz4fast-8",	ZIO_COMPRESS_LZ4FAST_8 },
+		{ "lz4fast-9",	ZIO_COMPRESS_LZ4FAST_9 },
+		{ "lz4fast-10",	ZIO_COMPRESS_LZ4FAST_10 },
+		{ "lz4fast-11",	ZIO_COMPRESS_LZ4FAST_11 },
+		{ "lz4fast-12",	ZIO_COMPRESS_LZ4FAST_12 },
+		{ "lz4fast-13",	ZIO_COMPRESS_LZ4FAST_13 },
+		{ "lz4fast-14",	ZIO_COMPRESS_LZ4FAST_14 },
+		{ "lz4fast-15",	ZIO_COMPRESS_LZ4FAST_15 },
+		{ "lz4fast-16",	ZIO_COMPRESS_LZ4FAST_16 },
+		{ "lz4fast-17",	ZIO_COMPRESS_LZ4FAST_17 },
+		{ "lz4fast-18",	ZIO_COMPRESS_LZ4FAST_18 },
+		{ "lz4fast-19",	ZIO_COMPRESS_LZ4FAST_19 },
+		{ "lz4fast-20",	ZIO_COMPRESS_LZ4FAST_20 },
+		{ "lz4fast-30",	ZIO_COMPRESS_LZ4FAST_30 },
+		{ "lz4fast-40",	ZIO_COMPRESS_LZ4FAST_40 },
+		{ "lz4fast-50",	ZIO_COMPRESS_LZ4FAST_50 },
+		{ "lz4fast-60",	ZIO_COMPRESS_LZ4FAST_60 },
+		{ "lz4fast-70",	ZIO_COMPRESS_LZ4FAST_70 },
+		{ "lz4fast-80",	ZIO_COMPRESS_LZ4FAST_80 },
+		{ "lz4fast-90",	ZIO_COMPRESS_LZ4FAST_90 },
+		{ "lz4fast-100",	ZIO_COMPRESS_LZ4FAST_100 },
```

Comment:
After N=20 I have only supported values in steps of 10 because of the low impact on compression ratio. I can extend the table when required. Keeping values up to 100 will give more flexibility when tuning `#define	COMPRESSIONLEVEL 12` for CPUs with big L1 cache.